Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -133,17 +133,19 @@ class TopSitesPanel: UIViewController {
         collection?.indexPathsForVisibleItems().forEach(updateRemoveButtonStateForIndexPath)
     }
 
-    private func deleteTileForSuggestedSite(site: SuggestedSite) -> Success {
+    private func deleteTileForSuggestedSite(site: Site) -> Success {
         var deletedSuggestedSites = profile.prefs.arrayForKey("topSites.deletedSuggestedSites") as! [String]
-        deletedSuggestedSites.append(site.url)
+        if !deletedSuggestedSites.contains(dataSource.extractDomainURL(site.url)) {
+            deletedSuggestedSites.append(site.url)
+        }
         profile.prefs.setObject(deletedSuggestedSites, forKey: "topSites.deletedSuggestedSites")
-        return succeed()
+        return self.profile.history.removeSiteFromTopSites(site)
     }
 
     private func deleteHistoryTileForSite(site: Site, atIndexPath indexPath: NSIndexPath) {
         collection?.userInteractionEnabled = false
 
-        let deletion = site is SuggestedSite ? deleteTileForSuggestedSite(site as! SuggestedSite) :  profile.history.removeSiteFromTopSites(site)
+        let deletion = site is SuggestedSite ? deleteTileForSuggestedSite(site) : profile.history.removeSiteFromTopSites(site)
```

Comment:
...and here, since you're already calling `profile.history.removeSiteFromTopSites(site)`, just do that unconditionally now.